mathics/builtin/files_io/filesystem.py

```diff
@@ -1359,7 +1362,14 @@ def eval_3(self, forms, paths, n, evaluation, **options):
         else:
             patterns = [
                 re.compile(
-                    "^" + to_regex(p, evaluation, abbreviated_patterns=True) + "$"
+                    "^"
+                    + to_regex(
```

Comment:
See comment above regarding `to_regex()`. 

Since this code is exactly the same as the "then" branch and the only difference is whether `re.IGNORECASE`  is used, I think it would be clearer if this the list comprehension were turned into an internal function which takes a boolean "ignore_case" parameter. 

Again, I realize this is a flaw of the original code.

Reply:
OK, if you have an idea about how it should be, and you are up for it, just commit the change over here.

Reply:
Ok - will do later.